wa/framework/workload.py

```diff
@@ -194,6 +156,76 @@ def __init__(self, target, **kwargs):
                                   force_install=kwargs.get('force_install'),
                                   install_timeout=kwargs.get('install_timeout'),
                                   uninstall=kwargs.get('uninstall'))
+
+    def init_resources(self, context):
+        pass
+
+    @once
```

Comment:
`initialize` should be used to resolve the APK/package (as only needs to happen once per run, rather than for each iteration) and so should not be done only `once` per run, but for each instance of the workload (as may specify different versions/variants).